### ISSUE TITLE
Enhancement:  Have Info On or Near "Declare Bankruptcy" button that this auto-sale (1830 and others)

### ISSUE THREAD

**Issue by rheretic** [tobymao/18xx#3861]
It is not clear what the Declare Bankruptcy button does.  My hope as the player is that I can avoid any painful stock sales and just give up.  

If Declare Bankruptcy actually automatically sells all my legal shares that should be shown.  

The system is rightfully calculating that I am insolvent.  The UI reports this in a blurb.  The next thing on the screen should be "Declare Bankruptcy" with **the explanation that this will sell all legal shares.**

**Comment by benjaminxscott** [comment]
tl;dr - it doesn't sell shares on your behalf, it's the same as clicking `End Game` in 1830

Declaring bankruptcy [doesn't sell shares or take other share actions](https://github.com/tobymao/18xx/blob/master/lib/engine/game/base.rb#L1248)

The button only shows up when a player is unable to raise money to cover liability (insolvent) during emergency money raising
(Note that this doesn't check whether the player could juggle trains, or buy a train across from another company)

The game doesn't know whether a player is trying to engineer bankruptcy to end the game, so it lets the player choose whether or not they are *really* insolvent

In 1830, declaring bankruptcy ends the game, which is [the default behavior](https://github.com/tobymao/18xx/blob/master/lib/engine/game/base.rb#L107) unless a game redefines it (some games reduce the cert limit when a player goes bankrupt, etc)

**Comment by rheretic** [comment]
Jen (I don't know her handle here) said on the Slack that this is exactly what this does (does a maximal share sale).

My understanding of the rules is that you must complete your legal share sales irrespective if you'll cover the cost or not.  _**AFTER**_ you do that if you don't have the money to buy the train then you are bankrupted and the game ends.

**Comment by jenf** [comment]
In 1830 is sells shares then ends the game.

**Comment by jenf** [comment]
https://github.com/tobymao/18xx/blob/master/lib/engine/step/bankrupt.rb

**Comment by rheretic** [comment]
Thanks Jen (and especially for the quick response!)

General comment that is not part of the PR:  I hate using unless like in https://github.com/tobymao/18xx/blob/master/lib/engine/step/bankrupt.rb#L28

I find the code very difficult to read.  (I like it as a Postfix Conditional but only then.)

**Comment by jenf** [comment]
We have code style enforced by rubocop.
